File: java-resilience4j/product/src/main/java/com/codegik/poc/javaresilience4j/controller/ChairController.java
package com.codegik.poc.javaresilience4j.controller;

import com.codegik.poc.javaresilience4j.domain.Chair;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RestController;

import java.util.Arrays;
import java.util.List;
import java.util.Optional;

@RestController("/chair")
public class ChairController {
    private final List<Chair> chairList = Arrays.asList(
            new Chair("1", "Table chair", 30, 50),
            new Chair("2", "Outside chair", 100, 80),
            new Chair("3", "Weird chair", 100, 20)
    );


    public ResponseEntity<List<Chair>> getAllChairs() {
        return ResponseEntity.ok(chairList);
    }


    public ResponseEntity<Optional<Chair>> getById(String id) {
        return ResponseEntity.ok(chairList.stream().filter(c -> id.equals(c.getId())).findFirst());
    }
}

File: kotlin-resilience4j/product/src/main/kotlin/com/codegik/poc/javaresilience4j/controller/ChairController.kt
package com.codegik.poc.javaresilience4j.controller

import com.codegik.poc.javaresilience4j.domain.Chair
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/chair")
class ChairController {
    private val chairList = listOf(
        Chair("1", "Table chair", 30, 50),
        Chair("2", "Outside chair", 100, 80),
        Chair("3", "Weird chair", 100, 20)
    )

    @GetMapping
    fun allChairs(): ResponseEntity<List<Chair>> {
        return ResponseEntity.ok(chairList)
    }

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: String): ResponseEntity<Chair> {
        return ResponseEntity.ok(chairList.filter { chair -> id == chair.id }.firstOrNull())
    }
}